File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[][]}
     */
    
    levelOrder(root) {
         if(root == null) return []
        
         let deque = new Deque();
         deque.pushBack(root)
    
         let result = []

         while(deque.size() > 0){
            let arr = []
            let len = deque.size()
            for(let i = 0; i < len; i++){
                let frontVal = deque.popFront()
                if(frontVal.left) deque.pushBack(frontVal.left)
                if(frontVal.right) deque.pushBack(frontVal.right)

                arr.push(frontVal.val);
            }
            result.push(arr)

         }  
        return result

    }
}

File: Data Structures & Algorithms/binary-tree-diameter/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:

    def diameterOfBinaryTree(self, root: Optional[TreeNode]) -> int:
        self.maximum = 0
        
        def recursion(root):
            if root == None:
                return -1
            left = recursion(root.left) + 1
            right = recursion(root.right) + 1
            print("val", root.val)
            print("left", left)
            print("right", right)
            if left + right > self.maximum:
                self.maximum = left + right
            print(self.maximum)
            return max(left, right)
        
        recursion(root)
        return self.maximum